test/batch_test.cc

```diff
@@ -0,0 +1,39 @@
+#include "benchmark/benchmark.h"
+
+#include <algorithm>
+
+#include "../src/check.h"  // NOTE: check.h is for internal use only!
+
+static void BM_batch_iteration_count(benchmark::State& state) {
+  const size_t batch_size = state.range(0);
+  size_t actual_iterations = 0;
+  while (state.KeepRunningBatch(batch_size)) {
+    actual_iterations += batch_size;
+  }
+  CHECK_GE(state.iterations(), state.max_iterations);
```

Comment:
Actually disregard me. There are no ODR violations when using `check.h` so this is fine. My mistake.